federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -724,18 +724,24 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	if !wantsDNSRecords(service.appliedState) {
 		return nil
 	}
-	ensuredCount := 0
+
+	var err error
+	missedCount := 0
+	unensuredCount := 0
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
-				ensuredCount += 1
+				if err = s.ensureDnsRecords(clusterName, service); err != nil {
+					unensuredCount += 1
+				}
+			} else {
+				missedCount += 1
 			}
 		}
 	}
-	if ensuredCount < len(clusterNames) {
-		return fmt.Errorf("Failed to update DNS records for %d of %d clusters for service %v due to missing clients for those clusters",
-			len(clusterNames)-ensuredCount, len(clusterNames), service)
+	if missedCount > 0 || unensuredCount > 0 {
+		return fmt.Errorf("Failed to update DNS records for %d clusters for service %v due to missing clients[missed count: %d] or ensuring DNS records error[unensured count: %d] %v",
+			len(clusterNames), service, missedCount, unensuredCount, err)
```

Comment:
Yes, agreed, although please make it clear in the error text that the reported error is only the last one.  Note also that in this code err might be nil (e.g. if all of the errors were misses, and none of them unensured.  So i think that you need to set err above in the case of missed.